### ISSUE TITLE
Crash when using fingerprint before OTP

### ISSUE THREAD

**Issue by PhilippC** [PhilippC/keepass2android#754]
when a database uses OTP Key provider, the app crashes when fingerprint is used before OTPs are entered.

**Comment by PhilippC** [comment]
seems fixed since 1.08

**Comment by Jaaaaanobert** [comment]
Hi,

The bug unfortunately still exists and was not fixed with version 1.08.

Scenario:
Database is already selected. 
I start the app, KeePass asks for my fingerprint. 
After pressing the fingerprint, the app crashes.

Second try:
I start the app and tap the "back" button of my phone to avoid pressing the fingerprint.
I manually press the field to read my YubiKey now and read it via NFC. 
After reading the YubiKey.
I manually press the fingerprint symbol next to the password-field and press the shown "unlock with fingerprint"-field to unlock the database.
The database opens without any issues or crashes.

The database is synced via ftp from a local NAS-Server and is accesses by Windows-Computers.  
I am using a YubiKey 5 NFC and the plugin "KeeChallenge" on the Windows machine to unlock the databases. 

Tested this on several devices with several databases (all stored on ftp-server) on android 10 and 11. I am using the newest KeePass version in the PlayStore (1.09c-r0). 
The bug was never fixed since opening this issue.

**Comment by pmorange** [comment]
Hi,
The issue is not fixed. 
Same as #1151 indeed.

Hope to see this fixed someday, but the issue has been present for so long that I don't know if I should still hope or make my peace with the issue...

Cheers.

**Comment by alexander-thiesen** [comment]
Hi, 
the issue is still not fixed. 
The only way it works is like described by @klausklemens in https://github.com/PhilippC/keepass2android/issues/1151#issuecomment-901753840_
"I always have to abort fingerprint reading, load otp auxilary file, and finally press the fingerprint button next to the password field and it works."

I wanted to try to fix the issue myself, but unfortunately, since I am inexperienced, I was unsuccessful to build keepass2android on my PC. 
My idea to solve the issue was just to disable fingerprint until the otp file was loaded and then prompt for fingerprint. Or disable automatic unlocking when the fingerprint is inserted when no otp file has been loaded. 

Cheers.

**Comment by mle-86** [comment]
Hi,

I have the same issue, on at least 3 different phones already (HMD Nokia, Asus and Samsung). For me it's not a big issue, but I wanted to add my voice here, to give feedback.
I also can not contribute code unfortunately. However, if it is a lot of work to fix the crash conditions, I feel like a simple settings toggle along the lines "do not auto-trigger fingerprint reader" would be a great workaround / solution.

Love the app, thanks for the great work!
Best regards

**Comment by mle-86** [comment]
Hi again,
After reading up a bit on the whole Yubikey Challenge Response thing, I found out that the KeeChallenge Plugin for KeePass 2 seems to be abandoned and also it's design has some negative security implications (see https://support.keepassium.com/kb/yubikey-keechallenge/#problems).
Which is why I decided to switch to the KeePassXC format with Yubico Challenge response, for which the workflow in KeePass2Android works perfectly (i.e. Fingerprint is asked for first, then, after successful biometric authentication, the YubiKey is queried and the database opens, without any cancelling needed nor app crashing).
So for myself I see this issue as no longer relevant, but of course I can not speak for the other users in this thread.

**Comment by mora-phi** [comment]
Indeed, also moved to KeePassXC both on desktop and mobile, and everything is fine.
